net/dns/resolver/macios_ext.go

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2021 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// +build darwin,ts_macext ios,ts_macext
+
+package resolver
+
+import (
+	"errors"
+	"net"
+
+	"tailscale.com/net/netns"
+	"tailscale.com/wgengine/monitor"
+)
+
+func init() {
+	initListenConfig = initListenConfigNetworkExtension
```

Comment:
Why do it like this rather than a file per platform like the rest of our code? No strong feeling from me.

Reply:
you mean making it a `func` rather than a func variable?

two reasons: can't check for `nil` at runtime if it's a func, and can't replace it in tests (as I did for Adrian in the latest rev)